### PR TITLE
Ruby-2339 Write Driver Query Cache documentation

### PR DESCRIPTION
**Pull request by gjchong25**

### CODE REVIEW

File: docs/tutorials/ruby-driver-crud-operations.txt [hunk 1]
```diff
@@ -130,6 +130,43 @@ object, or with ``Decimal128.from_string()``.
...
+
+  Mongo::QueryCache.enabled = true
+
+The following code shows how to enable the QueryCache within the context of a
+specific block:
+
+.. code-block:: ruby
+
+  Mongo::QueryCache.cache do
+  # all queries within this block are cached
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-crud-operations.txt:166]: This line should be indented 2 extra spaces.

File: docs/tutorials/ruby-driver-crud-operations.txt [hunk 2]
```diff
@@ -130,6 +130,43 @@ object, or with ``Decimal128.from_string()``.
    price = BSON::Decimal128.from_string("428.79")
    # => BSON::Decimal128('428.79')
 
+Query Cache
+===========
+
+If the Ruby driver's QueryCache is enabled, it will cache find queries on
```
Comment (p-mongo) [inline docs/tutorials/ruby-driver-crud-operations.txt:136]: And whenever "query cache" is used to refer to the concept and not specifically to the class in code, can you spell it as two words without special capitalization please.

File: docs/tutorials/ruby-driver-crud-operations.txt [hunk 3]
```diff
@@ -130,6 +130,44 @@ object, or with ``Decimal128.from_string()``.
    price = BSON::Decimal128.from_string("428.79")
    # => BSON::Decimal128('428.79')
 
+Query Cache
+===========
+
+If the Ruby driver's query cache is enabled, it will cache find queries on
+the currently executed thread and avoids sending requests to the database for
```
Comment (egiurleo) [inline docs/tutorials/ruby-driver-crud-operations.txt:137]: Small grammar thing -- I believe this should be `and avoid sending requests...`

File: docs/tutorials/ruby-driver-crud-operations.txt [hunk 4]
```diff
@@ -130,6 +130,44 @@ object, or with ``Decimal128.from_string()``.
...
+identical queries. The QueryCache class stores CachingCursor objects which attempt
+to load documents from memory before retrieving them from the database. Performing
+any write operations such as insert, update, or delete clears the query cache. Note that
+if the number of results is too large to be returned in a single batch,
+the query cache will not be used, even if Mongo::QueryCache.enabled is true, and
```
Comment (egiurleo) [inline docs/tutorials/ruby-driver-crud-operations.txt:142]: Could you make Mongo::QueryCache.enabled a code snippet with `` on each side?

**Pull request merged**